app/src/main/java/de/westnordost/streetcomplete/quests/localized_name/AddRoadName.kt

```diff
@@ -40,10 +40,18 @@ class AddRoadName(
     /** returns overpass query string for creating the quests */
     private fun getOverpassQuery(bbox: BoundingBox) =
         OverpassQLUtil.getGlobalOverpassBBox(bbox) +
-        ROADS_WITHOUT_NAMES + "; " +
+        ROADS_WITHOUT_NAMES + "->.unnamed;\n" +
+        "(\n" +
+        "  way.unnamed['access'!~'private|no'];\n" +
+        "  way.unnamed['foot']['foot'!~'private|no'];\n" +
```

Comment:
FYI `'` are only necessary if the key/value contains characters that are not a-z.

Also, use '^private|no$' for the regular expression, the regex operator in OQL ist a "find in", not a "matches whole string".

Reply:
I though about making it match solely `no` and `private` but I decided to make it match also stuff like `access=yes;no`.

I probably should look deeper into what is generated by quests with similar restrictions and follow this.

Reply:
And I added `'` because in past was burned by Overpass API deciding to start crashing on previously accepted but illegal syntax, so I prefer to be more careful (but I may also change it).